src/map/entities/battleentity.cpp

```diff
@@ -255,40 +255,44 @@ uint8 CBattleEntity::GetSpeed()
         return std::clamp<uint8>(outputSpeed, std::numeric_limits<uint8>::min(), std::numeric_limits<uint8>::max());
     }
 
-    // Flee, KIs, Gear penalties, Bolters Roll.
-    float additiveMods = static_cast<float>(getMod(Mod::MOVE_SPEED_STACKABLE)) / 100.0f;
+    // Gear penalties, Bolters Roll.
+    uint8 additiveMods = static_cast<uint8>(getMod(Mod::MOVE_SPEED_STACKABLE));
 
-    // Quickening and Mazurka. Only highest applies.
-    Mod modToUse = getMod(Mod::MOVE_SPEED_QUICKENING) > getMod(Mod::MOVE_SPEED_MAZURKA) ? Mod::MOVE_SPEED_QUICKENING : Mod::MOVE_SPEED_MAZURKA;
+    // Quickening and Mazurka. They share a cap. Additive.
+    uint8 effectAdditiveBonus = std::clamp<uint8>(getMod(Mod::MOVE_SPEED_QUICKENING) + getMod(Mod::MOVE_SPEED_MAZURKA), 0, 10);
 
-    float effectBonus = static_cast<float>(getMod(modToUse)) / 100.0f;
+    // Flee.
+    float fleeFactor = std::clamp<float>(1.0f + static_cast<float>(getMod(Mod::MOVE_SPEED_FLEE)) / 100.0f, 1.0f, 2.0f);
 
-    // Positive movement speed from gear. Only highest applies.
-    float gearBonus = 0.0f;
+    // Positive movement speed from gear and from Atmas. Only highest applies. Multiplicative to base speed.
+    float gearFactor = 1.0f;
 
     if (objtype == TYPE_PC)
     {
-        gearBonus = static_cast<float>(getMaxGearMod(Mod::MOVE_SPEED_GEAR_BONUS)) / 100.0f;
+        gearFactor = std::clamp<float>(1.0f + static_cast<float>(getMaxGearMod(Mod::MOVE_SPEED_GEAR_BONUS)) / 100.0f, 1.0f, 1.25f);
     }
 
     // Gravity and Curse. They seem additive to each other and the sum seems to be multiplicative.
-    float weightPenalties = static_cast<float>(getMod(Mod::MOVE_SPEED_WEIGHT_PENALTY)) / 100.0f;
+    float weightFactor = std::clamp<float>(1.0f - static_cast<float>(getMod(Mod::MOVE_SPEED_WEIGHT_PENALTY)) / 100.0f, 0.1f, 1.0f);
 
     // We have all the modifiers needed. Calculate final speed.
-    float modifiedSpeed = static_cast<float>(baseSpeed) * std::clamp<float>(1.0f + additiveMods + effectBonus, 0.1f, 1.6f) * (1.0f + gearBonus) * std::clamp<float>(1.0f - weightPenalties, 0.1f, 1.0f);
-
-    outputSpeed = static_cast<uint8>(modifiedSpeed);
-
-    // Set cap.
+    // This MUST BE DONE IN THIS ORDER. Using uint8 data type, we use that to floor.
+    outputSpeed = baseSpeed + additiveMods + effectAdditiveBonus;
```

Comment:
Think effectAdditiveBonus is applied after multipliers. Plumb boots + raptor mazurka + flee = 65, would be 70 if mazurka was affected by flee.

Reply:
I'm ready for another round of maths XD

Reply:
Just to add, effectAdditiveBonus is after flee bonus and after gear bonus. 20 speed + 24% gear = 24 speed, add chocobo mazurka brings it to 34 not 37.